packages/@lwc/engine-dom/src/apis/create-element.ts

```diff
@@ -104,7 +104,11 @@ export function createElement(
         );
     }
 
-    const UpgradableConstructor = getUpgradableConstructor(sel, renderer);
+    // tagName must be all lowercase, unfortunately, we have legacy code that is
+    // passing `sel` as a camel-case, which makes them invalid custom elements name
+    // the following line guarantees that this does not leaks beyond this point.
+    const tagName = StringToLowerCase.call(sel);
```

Comment:
we were doing this at the lower level in current code, now we lifted here, which is the only place in which this could be happening.